### PR TITLE
remove leading ~ from name when indexing metric names

### PR DESCRIPTION
**Pull request by replay**

### CODE REVIEW

File: docs/tags.rst [hunk 1]
```diff
@@ -17,7 +17,7 @@ Carbon will automatically decode the tags, normalize the tag order, and register
 
 .. _querying-tagged-series:
 
-Tag names must have a length >= 1 and may contain any ascii characters except ``;!^=``. Tag values must also have a length >= 1, and they may contain any ascii characters except ``;~``. UTF-8 characters may work for names and values, but they are not well tested and it is not recommended to use non-ascii characters in metric names or tags.
+Tag names must have a length >= 1 and may contain any ascii characters except ``;!^=``. Tag values must also have a length >= 1, they may contain any ascii characters except ``;`` and the first character must not be ``~``. UTF-8 characters may work for names and values, but they are not well tested and it is not recommended to use non-ascii characters in metric names or tags. Metric names get indexed under the special tag `name`, if a metric name starts with one or multiple `~` they simply get removed from the according tag value because the `~` character is not allowed to be in the first position of the tag value. If a metric name consists of no other characters than `~`, then it is considered invalid and may get dropped.
```
Comment (ploxiln) [inline docs/tags.rst]: "removed from the according tag value"

"according" should be "corresponding" or (probably better) "derived"

Comment (replay) [reply docs/tags.rst]: thx, will fix that

**Comment by DanCech** [comment]
We want to keep TaggedSeries the same between graphite-web and carbon (ideally we'd break it out into a separate graphite-util package but if we're going to move toward a go implementation of carbon it doesn't make sense to put in the work required to do that vs just keeping the class in sync across the repos).  If you can sync them up that would be great, I'd also suggest that the sanitize function should use https://docs.python.org/2/library/string.html#string.lstrip vs replace.

**Comment by replay** [comment]
> I'd also suggest that the sanitize function should use https://docs.python.org/2/library/string.html#string.lstrip vs replace

I'm not sure which sanitize function you mean, the one in the carbon project, right? If so, then I've already updated that to use `.lstrip` in https://github.com/graphite-project/carbon/pull/858/files
Or do you think that graphite should also sanitize the values when it receives a request to tag a series? So far graphite is only validating, but not sanitizing. Carbon is doing the sanitizing.

**Comment by replay** [comment]
I didn't realize that the `TaggedSeries` class in carbon and graphite is pretty much the same, i'll sync the carbon one to graphite then

**Comment by replay** [comment]
I've pushed a few more tests, so the test coverage report is happy

**Pull request merged**